### DISCUSSION TITLE
Add clipboard support to textboxes in mobile platforms

### DISCUSSION THREAD

**Discussion by Mrcubix** [ppy/osu#15346] [Q&A] [answered]
Like the title state, i cannot copy paste text in lazer on android.
- Holding down on the text field will not show the android's context menu where you can tap on paste.
- Tapping on clipboard will bring you back to the keyboard instead.

OS: Android 8.0
Build: 2021.1028.0

**Comment by peppy** [comment]
You can use an external keyboard for now.

**Comment by Mrcubix** [comment]
Well i don't have an external keyboard on hand right now and hacker's keyboard which has a ctrl key that can be used to paste text doesn't work.

So i'll take a bet that you cannot paste text in lazer on android.
Guess i'll have to type it manually later since it's required to be logged in to use osu!direct and manually importing them is just obnoxious when the app restart every single time, but that's a known issue.

**Reply by peppy** [reply]
On iOS you can use a three finger gesture to paste - i'm not sure if that exists on android but may be possible if so.

**Reply by Mrcubix** [reply]
Gesture stuff heavily depend on the device and the manufacturer, it's not a thing on my samsung j6 and is probably not a thing on high end phones

Anyway, here are my options on my device outside of using third party apps

Context menu, with, copy, paste, select all:
![20211028_133228](https://user-images.githubusercontent.com/39861216/139247768-3db7b86f-1c55-4bd2-81d4-82470d2ac911.jpg)


Clipboard:
![20211028_133241](https://user-images.githubusercontent.com/39861216/139247825-3a2f0c32-0629-4f8e-8647-fb2b37a136ab.jpg)


As a reminder, both of these cannot be used in lazer.

As for third party keyboards, i tried "Haker's keyboard which allow me to ctrl+c and crtl + v, but that also doesn't work in lazer.

I didn't try any other thrid party apps as of now.

**Comment by Susko3** [comment, marked as answer]
https://github.com/ppy/osu/pull/24184
